tests/nvapi.cpp

```diff
@@ -0,0 +1,9 @@
+#include "../inc/catch.hpp"
+#include "../inc/catch2/trompeloeil.hpp"
+#include "../src/nvapi.cpp"
+
+TEST_CASE("Initialize and unloads") {
```

Comment:
Yeah, this is work in progress.
What i would love to do here, is mocking all our boundaries (`DXGIFactory`, `Vulkan` and `Nvml`) to run the tests without any external dependencies and with different scenarios. I thought about introducing a  kind of `ResourceFactory` which creates above classes and providing an overload for `Initialize` to accept an alternative resource factory that returns only mocked instances. But not yet sure how this works out, esp. with the `unique_ptr`'s.

Reply:
Sounds like a good approach.

> But not yet sure how this works out, esp. with the unique_ptr's.

We can either return raw pointers and have intended owners take ownership of them by creating `unique_ptr` from them on their own or just return `unique_ptr` directly have copy ellision do its magic.

Reply:
I have added a resource factory and use a different implementation for testing which just returns mocks  Actually the handing of the `unique_ptr` looks quite ok imho.
While writing this, calling the create methods in the alternative implementation twice will probably go wrong since the `unique_ptr` are reset after moving them in the first call. Well, lets see how this turns out with more tests. Best conclusion would be to just assert in that situation since I would like to avoid sharing state across tests.